### PR TITLE
Reco commands risk breakdown

### PR DESCRIPTION
**Pull request by GalNakash-RecoLabs**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Reco/Integrations/Reco/Reco.py [hunk 1]
```diff
@@ -35,6 +36,21 @@
 STEP_INIT = "init"
 
 
+def extract_response(response: Any) -> List[Dict[str, Any]]:
```
Comment (michal-dagan) [inline Packs/Reco/Integrations/Reco/Reco.py:39]: You can use the helper function also in - lines 619-630

File: Packs/Reco/Integrations/Reco/Reco.py [hunk 2]
```diff
@@ -35,6 +36,21 @@
...
+    if response.get("getTableResponse") is None:
+        demisto.error(f"got bad response, {response}")
+        raise Exception(f"got bad response, {response}")
+    else:
+        demisto.info(
+            f"Count of assets exposed publicly: {response.get('getTableResponse').get('totalNumberOfResults')}"
```
Comment (michal-dagan) [inline Packs/Reco/Integrations/Reco/Reco.py:45]: All the uses of the function are of "assets exposed publicly?"
```suggestion
            f"Count of assets: {response.get('getTableResponse').get('totalNumberOfResults')}"
```

**Review by michal-dagan — COMMENTED** [review]
Hi @GalNakash-RecoLabs,
Thank you for your contribution!
Great work!
Please see me comments.
Also, there are a few validation issues that need addressing:
1. The following commands appear in Reco.yml but not in the README file:
	reco-get-3rd-parties-accessible-to-data-list
	reco-get-sensitive-assets-with-public-link
	reco-get-files-shared-with-3rd-parties

Please generate a README.md file for your integration/script.
To do that, please run the `demisto-sdk generate-docs` command (use the -c flag for using a comma-separated command names to generate doc for) - for more information please see: https://xsoar.pan.dev/docs/concepts/demisto-sdk#generate-docs.

2. Reco_Reduce_Risk_Google_Publicly_Exposed_Files playbook - 
try running `demisto-sdk format -i Packs/Reco/Playbooks/Reco_Reduce_Risk_Google_Publicly_Exposed_Files.yml`

3. Please add a unit-test as the build fails on test-coverage:
https://app.circleci.com/pipelines/github/demisto/content/259430/workflows/2f39f4de-9346-474a-b5cf-69d34ec84399/jobs/700664/parallel-runs/0/steps/0-121


Please feel free to reach out to me with any questions - I'm available here or on slack :)
Thanks again

File: Packs/Reco/ReleaseNotes/1_1_1.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+
+#### Integrations
+
+##### Reco
+
+- Get sensitive assets publicly exposed 
+- Get files shared with 3rd parties 
+- reco-get-sensitive-assets-with-public-link
+- reco-get-3rd-parties-accessible-to-data-list
+- reco-get-files-shared-with-3rd-parties
```
Comment (michal-dagan) [inline Packs/Reco/ReleaseNotes/1_1_1.md:10]:
```suggestion
Added the following commands:
- reco-get-sensitive-assets-with-public-link
- reco-get-3rd-parties-accessible-to-data-list
- reco-get-files-shared-with-3rd-parties
```

File: Packs/Reco/ReleaseNotes/1_1_1.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+#### Playbooks
+
+##### New: Reco-Reduce-Risk-Google-Publicly-Exposed-Files
```
Comment (michal-dagan) [inline Packs/Reco/ReleaseNotes/1_1_1.md:14]: The playbook name is "Reco - Reduce Risk - Google Publicly Exposed Files"
please update

File: Packs/Reco/Playbooks/Reco_Reduce_Risk_Google_Publicly_Exposed_Files.yml [hunk 5]
```diff
@@ -0,0 +1,419 @@
...
+    isautoswitchedtoquietmode: false
+  "2":
+    id: "2"
+    taskid: 621d18d2-3ebf-4025-8509-dee865d92efe
+    type: condition
+    task:
+      id: 621d18d2-3ebf-4025-8509-dee865d92efe
+      version: -1
+      name: Location is User Drive
+      type: condition
```
[the remark this answers is not in the document]
Comment (michal-dagan) [reply Packs/Reco/Playbooks/Reco_Reduce_Risk_Google_Publicly_Exposed_Files.yml:77]: Please update :)
'[ERROR]: Packs/Reco/Playbooks/Reco_Reduce_Risk_Google_Publicly_Exposed_Files.yml: [ST107] - Missing the field "description" in Path: 'tasks'-> '2'-> 'task'
Please add the field "description" to the path:  'tasks'-> '2'-> 'task' in the yml.'

**Review by michal-dagan — COMMENTED** [review]
@GalNakash-RecoLabs,
Nice work! Added a few minor comments to fix validation
* The build still fails on [coverage](https://app.circleci.com/pipelines/github/demisto/content/259501/workflows/a8570173-249c-4729-a897-13c45b70d71c/jobs/700720/parallel-runs/0/steps/0-121), 
please add a unit-test, maybe try to simulate an Exception in the new functions you’ve added.
I'm available for any of your questions here or on slack.
Thanks again.

**Comment by GalNakash-RecoLabs** [comment]
@michal-dagan done!

File: Packs/Reco/Integrations/Reco/Reco.py [hunk 6]
```diff
@@ -942,102 +1147,117 @@ def main() -> None:
...
-        elif command == "reco-get-link-to-user-overview-page":
-            result = get_link_to_user_overview_page(
-                reco_client,
-                demisto.args()["entity"],
-                demisto.args()["param"],
-            )
-            return_results(result)
-        else:
-            raise NotImplementedError(f"{command} is not an existing reco command")
+        command_handler(command, max_fetch, params, reco_client)
```
[the remark this answers is not in the document]
Comment (GalNakash-RecoLabs) [reply Packs/Reco/Integrations/Reco/Reco.py]: Done

**Comment by efelmandar** [comment]
Hi @GalNakash-RecoLabs Thank you for your contribution, I reviewed the playbook and it looks great! My only suggestion is to provide descriptions for the playbook input, and tasks, or even expand the playbook's description itself a bit to help the users understand and use the playbook and inputs correctly. besides that, I think it's a good use case, nice work!

**Comment by GalNakash-RecoLabs** [comment]
> Hi @GalNakash-RecoLabs Thank you for your contribution, I reviewed the playbook and it looks great! My only suggestion is to provide descriptions for the playbook input, and tasks, or even expand the playbook's description itself a bit to help the users understand and use the playbook and inputs correctly. besides that, I think it's a good use case, nice work!

Thanks @efelmandar  - fixed everything :) let's merge

**Comment by michal-dagan** [comment]
@GalNakash-RecoLabs 
About the demo - let me know whether you want to record and share with me a demo video, or if you want to schedule a meeting for a live demo, whichever works best for you.
thanks

**Comment by GalNakash-RecoLabs** [comment]
> @GalNakash-RecoLabs About the demo - let me know whether you want to record and share with me a demo video, or if you want to schedule a meeting for a live demo, whichever works best for you. thanks

Done, sent video over Slack

File: Packs/Reco/ReleaseNotes/1_1_0.md [hunk 7]
```diff
@@ -5,4 +5,4 @@
 
 - Get new type of alerts
 - Get magic link to Reco portal
-- Updated the Docker image to: *demisto/python3:3.10.11.58677*.
+- Updated the Docker image to: demisto/python3:3.10.11.59581
```
Comment (michal-dagan) [inline Packs/Reco/ReleaseNotes/1_1_0.md:8]: please revert this change :)
you should have only one releaseNote file for each pr

File: Packs/Reco/ReleaseNotes/1_1_1.md [hunk 8]
```diff
@@ -0,0 +1,15 @@
+
+#### Integrations
+
+##### Reco
+
+Added the following commands:
+- reco-get-sensitive-assets-with-public-link
+- reco-get-3rd-parties-accessible-to-data-list
+- reco-get-files-shared-with-3rd-parties
```
Comment (michal-dagan) [inline Packs/Reco/ReleaseNotes/1_1_1.md:9]:
```suggestion
- Added the following commands:
    - ***reco-get-sensitive-assets-with-public-link***
    - ***reco-get-3rd-parties-accessible-to-data-list***
    - ***reco-get-files-shared-with-3rd-parties***
- Updated the Docker image to: *demisto/python3:3.10.11.59581*.

```

**Review by michal-dagan — COMMENTED** [review]
@GalNakash-RecoLabs 
Great work, we are almost done here...
There are some last issues to address before we merge this PR:
- please fix the rn
- As the [validation](https://app.circleci.com/pipelines/github/demisto/content/261526/workflows/94cb0e1a-e009-4bcc-8f57-6cd92498ff6a/jobs/701738/parallel-runs/0/steps/0-119) step failed, it looks like something changed in the playbook:
The following should be fixed:
first please run again  `demisto-sdk format -i Packs/Reco/Playbooks/Reco_Reduce_Risk_Google_Publicly_Exposed_Files.yml `

```
[ERROR]: Packs/Reco/Playbooks/Reco_Reduce_Risk_Google_Publicly_Exposed_Files.yml: [ST109] - Missing the field "fromversion" in root
Please add to the root of the yml.

[ERROR]: Packs/Reco/Playbooks/Reco_Reduce_Risk_Google_Publicly_Exposed_Files.yml: [ST108] - The field "vcShouldKeepItemLegacyProdMachine" was not defined in the scheme
To fix the problem, try running `demisto-sdk format -i Packs/Reco/Playbooks/Reco_Reduce_Risk_Google_Publicly_Exposed_Files.yml `

[ERROR]: Packs/Reco/Playbooks/Reco_Reduce_Risk_Google_Publicly_Exposed_Files.yml: [ST107] - Missing the field "description" in Path: 'tasks'-> '0'-> 'task'
Please add the field "description" to the path:  'tasks'-> '0'-> 'task' in the yml.

[ERROR]: Packs/Reco/Playbooks/Reco_Reduce_Risk_Google_Publicly_Exposed_Files.yml: [ST107] - Missing the field "description" in Path: 'tasks'-> '10'-> 'task'
Please add the field "description" to the path:  'tasks'-> '10'-> 'task' in the yml.

[ERROR]: Packs/Reco/Playbooks/Reco_Reduce_Risk_Google_Publicly_Exposed_Files.yml: [ST107] - Missing the field "description" in Path: 'tasks'-> '11'-> 'task'
Please add the field "description" to the path:  'tasks'-> '11'-> 'task' in the yml.

```

**Comment by GalNakash-RecoLabs** [comment]
> 

Done

> @GalNakash-RecoLabs Great work, we are almost done here... There are some last issues to address before we merge this PR:
> 
> * please fix the rn
> * As the [validation](https://app.circleci.com/pipelines/github/demisto/content/261526/workflows/94cb0e1a-e009-4bcc-8f57-6cd92498ff6a/jobs/701738/parallel-runs/0/steps/0-119) step failed, it looks like something changed in the playbook:
>   The following should be fixed:
>   first please run again  `demisto-sdk format -i Packs/Reco/Playbooks/Reco_Reduce_Risk_Google_Publicly_Exposed_Files.yml `
> 
> ```
> [ERROR]: Packs/Reco/Playbooks/Reco_Reduce_Risk_Google_Publicly_Exposed_Files.yml: [ST109] - Missing the field "fromversion" in root
> Please add to the root of the yml.
> 
> [ERROR]: Packs/Reco/Playbooks/Reco_Reduce_Risk_Google_Publicly_Exposed_Files.yml: [ST108] - The field "vcShouldKeepItemLegacyProdMachine" was not defined in the scheme
> To fix the problem, try running `demisto-sdk format -i Packs/Reco/Playbooks/Reco_Reduce_Risk_Google_Publicly_Exposed_Files.yml `
> 
> [ERROR]: Packs/Reco/Playbooks/Reco_Reduce_Risk_Google_Publicly_Exposed_Files.yml: [ST107] - Missing the field "description" in Path: 'tasks'-> '0'-> 'task'
> Please add the field "description" to the path:  'tasks'-> '0'-> 'task' in the yml.
> 
> [ERROR]: Packs/Reco/Playbooks/Reco_Reduce_Risk_Google_Publicly_Exposed_Files.yml: [ST107] - Missing the field "description" in Path: 'tasks'-> '10'-> 'task'
> Please add the field "description" to the path:  'tasks'-> '10'-> 'task' in the yml.
> 
> [ERROR]: Packs/Reco/Playbooks/Reco_Reduce_Risk_Google_Publicly_Exposed_Files.yml: [ST107] - Missing the field "description" in Path: 'tasks'-> '11'-> 'task'
> Please add the field "description" to the path:  'tasks'-> '11'-> 'task' in the yml.
> ```

DONE

**Review by michal-dagan — APPROVED** [review]
